### ISSUE TITLE
Lineage 17.1 Open Gapps and Andoid Auto

### ISSUE THREAD

**Issue by ghost** [opengapps/opengapps#894]
I  instale the ( Lineage os 17.1.20200415 and Gapps) on my S8+ . All ok ,Thanks to the developer . The only problem is with Android Auto and the usb connection in Lineage software... maybe authorities usb choice is not to "file tranfer"...
In S8+ with original samsung os "android 9" the application android auto with same version as above function very well.
I think the problem is the way that lineage manage the usb connection with Android Auto!!!
When Android Auto strat in the infotainment the phone usb force (goes) to "without file transfer" and the connection stops
Any help
![20201204_080831](https://user-images.githubusercontent.com/75583308/101289466-54956180-3805-11eb-8f2c-ad269241ed6c.jpg)

**Comment by huw0** [comment]
Think this may be the same cause as #742

**Comment by jameshilliard** [comment]
Possibly related boot issue I'm seeing:
```
12-13 11:45:36.462  5602  5602 E System  : ******************************************
12-13 11:45:36.462  5602  5602 E System  : ************ Failure starting system services
12-13 11:45:36.462  5602  5602 E System  : java.lang.IllegalStateException: Signature|privileged permissions not in privapp-permissions whitelist: {com.google.android.projection.gearhead: android.permission.CHANGE_COMPONENT_ENABLED_STATE}
12-13 11:45:36.462  5602  5602 E System  : 	at com.android.server.pm.permission.PermissionManagerService.systemReady(PermissionManagerService.java:2970)
12-13 11:45:36.462  5602  5602 E System  : 	at com.android.server.pm.permission.PermissionManagerService.access$100(PermissionManagerService.java:122)
12-13 11:45:36.462  5602  5602 E System  : 	at com.android.server.pm.permission.PermissionManagerService$PermissionManagerServiceInternalImpl.systemReady(PermissionManagerService.java:3031)
12-13 11:45:36.462  5602  5602 E System  : 	at com.android.server.pm.PackageManagerService.systemReady(PackageManagerService.java:21935)
12-13 11:45:36.462  5602  5602 E System  : 	at com.android.server.SystemServer.startOtherServices(SystemServer.java:2037)
12-13 11:45:36.462  5602  5602 E System  : 	at com.android.server.SystemServer.run(SystemServer.java:529)
12-13 11:45:36.462  5602  5602 E System  : 	at com.android.server.SystemServer.main(SystemServer.java:356)
12-13 11:45:36.462  5602  5602 E System  : 	at java.lang.reflect.Method.invoke(Native Method)
12-13 11:45:36.462  5602  5602 E System  : 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
12-13 11:45:36.462  5602  5602 E System  : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:918)
12-13 11:45:36.462  5602  5602 D SystemServerTiming: MakePackageManagerServiceReady took to complete: 850ms
12-13 11:45:36.463  5602  5602 E Zygote  : System zygote died with exception
12-13 11:45:36.463  5602  5602 E Zygote  : java.lang.IllegalStateException: Signature|privileged permissions not in privapp-permissions whitelist: {com.google.android.projection.gearhead: android.permission.CHANGE_COMPONENT_ENABLED_STATE}
12-13 11:45:36.463  5602  5602 E Zygote  : 	at com.android.server.pm.permission.PermissionManagerService.systemReady(PermissionManagerService.java:2970)
12-13 11:45:36.463  5602  5602 E Zygote  : 	at com.android.server.pm.permission.PermissionManagerService.access$100(PermissionManagerService.java:122)
12-13 11:45:36.463  5602  5602 E Zygote  : 	at com.android.server.pm.permission.PermissionManagerService$PermissionManagerServiceInternalImpl.systemReady(PermissionManagerService.java:3031)
12-13 11:45:36.463  5602  5602 E Zygote  : 	at com.android.server.pm.PackageManagerService.systemReady(PackageManagerService.java:21935)
12-13 11:45:36.463  5602  5602 E Zygote  : 	at com.android.server.SystemServer.startOtherServices(SystemServer.java:2037)
12-13 11:45:36.463  5602  5602 E Zygote  : 	at com.android.server.SystemServer.run(SystemServer.java:529)
12-13 11:45:36.463  5602  5602 E Zygote  : 	at com.android.server.SystemServer.main(SystemServer.java:356)
12-13 11:45:36.463  5602  5602 E Zygote  : 	at java.lang.reflect.Method.invoke(Native Method)
12-13 11:45:36.463  5602  5602 E Zygote  : 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
12-13 11:45:36.463  5602  5602 E Zygote  : 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:918)
12-13 11:45:36.463  5602  5602 D AndroidRuntime: Shutting down VM
12-13 11:45:36.463  5602  5602 E AndroidRuntime: *** FATAL EXCEPTION IN SYSTEM PROCESS: main
12-13 11:45:36.463  5602  5602 E AndroidRuntime: java.lang.IllegalStateException: Signature|privileged permissions not in privapp-permissions whitelist: {com.google.android.projection.gearhead: android.permission.CHANGE_COMPONENT_ENABLED_STATE}
12-13 11:45:36.463  5602  5602 E AndroidRuntime: 	at com.android.server.pm.permission.PermissionManagerService.systemReady(PermissionManagerService.java:2970)
12-13 11:45:36.463  5602  5602 E AndroidRuntime: 	at com.android.server.pm.permission.PermissionManagerService.access$100(PermissionManagerService.java:122)
12-13 11:45:36.463  5602  5602 E AndroidRuntime: 	at com.android.server.pm.permission.PermissionManagerService$PermissionManagerServiceInternalImpl.systemReady(PermissionManagerService.java:3031)
12-13 11:45:36.463  5602  5602 E AndroidRuntime: 	at com.android.server.pm.PackageManagerService.systemReady(PackageManagerService.java:21935)
12-13 11:45:36.463  5602  5602 E AndroidRuntime: 	at com.android.server.SystemServer.startOtherServices(SystemServer.java:2037)
12-13 11:45:36.463  5602  5602 E AndroidRuntime: 	at com.android.server.SystemServer.run(SystemServer.java:529)
12-13 11:45:36.463  5602  5602 E AndroidRuntime: 	at com.android.server.SystemServer.main(SystemServer.java:356)
12-13 11:45:36.463  5602  5602 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
12-13 11:45:36.463  5602  5602 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
12-13 11:45:36.463  5602  5602 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:918)
```

**Comment by yung-wang** [comment]
Im getting similiar problems with opengapps pico after version open_gapps-arm64-10.0-pico-20201121.zip. Newer versions are not working.

When I open android auto on the phone, i get the message: requires one or more google play services.
According this post, some apps requires to be installed as system. But it has always been working since 20201121.
https://forum.xda-developers.com/t/rom-11-0-unofficial-lineageos-18-1.4173997/post-84062007

**Comment by yung-wang** [comment]
Still having problems with opengapps. I switch to nikgapps core + android auto addon, this works fine.

**Comment by jacob-pro** [comment]
@yung-wang  did you have to do anything to remove opengapps first or did you just flash nikgapps?

**Comment by yung-wang** [comment]
@jacob-pro I wiped system and cache first. Than flashed rom + nikgapps + addon. That is all 😃.

**Comment by osm0sis** [comment]
@ghost Is this a duplicate of #742?

**Comment by osm0sis** [comment]
That permission is already present here: https://gitlab.opengapps.org/opengapps/all/-/blob/master/permissions/privapp-permissions-google.xml#L670

**Comment by KeepUrDayJob** [comment]
> @jacob-pro I wiped system and cache first. Than flashed rom + nikgapps + addon. That is all 😃.

This worked for me too, thank you!!

**Comment by yung-wang** [comment]
You're welcome. I see they are fixing it. Maybe it should work with the latest opengapps? I didnt test it.

**Comment by ronaldwan** [comment]
Just tried today's opengapps build - both pico and stock - looks like its still failing.

I can confirm that nikgapps core + android auto from 02/24 does work though.

**Comment by agross** [comment]
@ronaldwan 

> I can confirm that nikgapps core + **android auto from 02/24** does work though.

Are you referring to the nikgapps addon or the app from the Play Store?

**Comment by yung-wang** [comment]
I use the addons, not from the play store.
You can download them here:
https://sourceforge.net/projects/nikgapps/files/Releases/

Android 11

Core:
https://sourceforge.net/projects/nikgapps/files/Releases/NikGapps-R/30-Jan-2021/NikGapps-core-arm64-11-20210130-signed.zip/download

Addons:
https://sourceforge.net/projects/nikgapps/files/Releases/Addons-R/24-Jan-2021/NikGapps-Addon-11-AndroidAuto-signed.zip/download

**Comment by ronaldwan** [comment]
> @ronaldwan
> 
> > I can confirm that nikgapps core + **android auto from 02/24** does work though.
> 
> Are you referring to the nikgapps addon or the app from the Play Store?

I used the android auto add-on. I can confirm subsequent updates from the App Store also works.

I'm not familiar with the Opengapps build system. Are the Opengapps downloadable builds nightly or from some sort of release branch?

**Comment by agross** [comment]
AFAIK OpenGapps are updated daily (https://opengapps.org/). I tested today's nano version with Android Auto from Play Store. Didn't work.

I then switched to nikgapps core (just flashed from Recovery) without any add-ons and Android Auto from Play store. It finally is able to start without displaying "Something went wrong". Will have to check the car connection, though.

**Comment by zundertj** [comment]
In case this might be useful, I had a similar problem where after flashing my phone with the latest software (instead of the standard OnePlus software) AA no longer worked, USB transfer type was on connect immediately reset to "no transfer" similar to what the first post says here.

Android: 10
Phone: OnePlus3
LineageOS: lineage-17.1-20210222-nightly-oneplus3-signed
GApps: open_gapps-arm64-10.0-micro-20210227

Following https://forum.xda-developers.com/t/gapps-daily-open-gapps-for-android-all-android-versions-devices.3098071/page-351#post-84511137, I decided to try `open_gapps-arm64-10.0-micro-20210206` and that worked. I did need to install Android Auto app (and the app store did not allow me to do this due to location restrictions, got an apk instead) because that is probably not on the micro version. So my hypothesis would be that there is an issue introduced between 20210206 and 20210227 issues.

The one small issue that remains is that on connect the display in my car turns white for two seconds before displaying the AA interface, it didnt do that with the standard OP software.

**Comment by agross** [comment]
> Will have to check the car connection, though.

It worked!

**Comment by anirudhb** [comment]
Had the same issue, can confirm `open_gapps-arm-10.0-mini-20210206` works! :)

**Comment by ElSabio97** [comment]
> Had the same issue, can confirm `open_gapps-arm-10.0-mini-20210206` works! :)

Do we know on what release it does stop working? and if its gonna ever work on future releases?

**Comment by yung-wang** [comment]
Too bad pico does not work. It worked before. I like to have a very minimal gapps. 
Nice to hear mini works.

**Comment by sanomaster** [comment]
It works for me with BiTGApps R20. It supports arm (32) bit.

**Comment by ElSabio97** [comment]
> Had the same issue, can confirm `open_gapps-arm-10.0-mini-20210206` works! :)

I can confirm this one works great for me too! (One plus One, LOS 17.1)

**Comment by Alex9001** [comment]
March 25th mini does not work with Android Auto. Reverting to open_gapps-arm64-10.0-pico-20210218 as that version works.

Nikgapps, unfortunately, is lagged for notifications for Signal and Voice.

**Comment by taylorkline** [comment]
Is anyone else seeing this message about "No installed apps work with this USB accessory"?

![Screenshot_20210326-114940_System_UI](https://user-images.githubusercontent.com/2641739/112665733-5e30ec80-8e29-11eb-82ae-c4caca8a0bbe.png)

**Comment by sanomaster** [comment]
> Is anyone else seeing this message about "No installed apps work with this USB accessory"?
> 
> ![Screenshot_20210326-114940_System_UI](https://user-images.githubusercontent.com/2641739/112665733-5e30ec80-8e29-11eb-82ae-c4caca8a0bbe.png)

Yes, when connecting the phone to car's head unit.
Try to install open_gapps-arm-10.0-mini-20210206

**Comment by yung-wang** [comment]
For lineageos 18.1 im using MindTheGapps-11.0.0-arm64-20210412_124247.zip. This works fine.
https://wiki.lineageos.org/gapps.html

**Comment by osm0sis** [comment]
Resolved fully in 61c1088 I believe 👍
